### ISSUE TITLE
[Serializer] BC break on isers following #54148

### ISSUE THREAD

**Issue by ninsuo** [symfony/symfony#54488]
### Symfony version(s) affected

6.4.6

### Description

The following PR fixes a bug, but changes how isers are serialized: https://github.com/symfony/symfony/pull/54148/files

- In 6.4.4, a property named `$isSomething` is serialized `something`

- In 6.4.6, that property is now serialized: `isSomething`

Even though previous behaviour was a bug, it is there since years and needs the right migration on our codebase. I'd advise that such a backward compatibility break could be updated on the next major upgrade.

### How to reproduce

```php
    private bool $isAdmin = false;

    public function isAdmin(): bool
    {
        return $this->isAdmin;
    }

    public function setIsAdmin(bool $isAdmin): self
    {
        $this->isAdmin = $isAdmin;

        return $this;
    }
```

**Comment by mtarld** [comment]
This is related to the following PR: https://github.com/symfony/symfony/pull/54148

**Comment by mtarld** [comment]
Indeed, I think that https://github.com/symfony/symfony/pull/54148 should be reverted and merged in 7.2 instead.

**Comment by henrythethird** [comment]
The `5.4.38` Version is affected as well. `5.4.36` was not yet affected (we rolled back to that one)

This introduces a backward compatibility issue. Wouldn't this be a breaking change to the serializer component?

Our current workaround (maybe useful for someone):
```
    "extra": {
        "symfony": {
            "require": "5.4.*,!=5.4.38"
        }
    }
```
